datastore/importers/nmdb-import-hosts/Parser.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright 2017 National Technology & Engineering Solutions of Sandia, LLC
+// (NTESS). Under the terms of Contract DE-NA0003525 with NTESS, the U.S.
+// Government retains certain rights in this software.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+// =============================================================================
+// Maintained by Sandia National Laboratories <Netmeld@sandia.gov>
+// =============================================================================
+
+/* Notes:
+   - This unit is part of the complilation process to help ensure consistency
+     between templates and the actual data
+   - Various data is included and most is commented solely for educational
+     purposes
+     - In non-template, remove data as makes sense
+
+   Guidelines:
+   - If using a custom Parser
+     - Data ordering is different as the focus is the parsing logic, not rule
+       instantiation
+     - It occasionally is more reasonable to interact and place data with an
+       intermediary object
+       - The code can be collocated or a separate file, depending on complexity
+*/
+
+#include "Parser.hpp"
+
+// =============================================================================
+// Parser logic
+// =============================================================================
+Parser::Parser() : Parser::base_type(start)
+{
+  start =
+    *(line)
+    ;
+
+  // (IP address [save to _val] >> domain name [add an alias for the IP in _val that is the result of the parsed ParserDomainName with the reason "hosts file"] >> not a comment >> end of line) | (comment >> end of line) | (end of line)
+  line = (ipAddr [(qi::_val = qi::_1)] >> +domainName[(pnx::bind(&nmdo::IpAddress::addAlias, &qi::_val, qi::_1, "hosts file"))] >> -comment >> qi::eol) | (comment >> qi::eol) | (qi::eol);
+
+  comment = (qi::lit('#') >> *(qi::ascii::graph | qi::ascii::blank));
```

Comment:
Similar to prior rollback comment.

```suggestion
  comment = (qi::lit('#') > *(qi::ascii::graph | qi::ascii::blank));
```